src/hooks/usePersonalData/usePersonalData.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useReducer, useContext, useCallback, Dispatch } from 'react'
+
+import {
+  FollowedChannel,
+  WatchedVideo,
+  PersonalDataClient,
+  getInitialPersonalData,
+  localStorageClient,
+} from './localStorageClient'
+
+const personalClient = localStorageClient
+
+type UpdateWatchedVideosAction = {
+  type: 'UPDATE_WATCHED_VIDEOS'
+  watchedVideos: WatchedVideo[]
+}
+type UpdateChannelFollowingAction = {
+  type: 'UPDATE_FOLLOWED_CHANNELS'
+  followedChannels: FollowedChannel[]
+}
+
+type Action = UpdateWatchedVideosAction | UpdateChannelFollowingAction
+function asyncReducer(state: PersonalData, action: Action): PersonalData {
+  switch (action.type) {
+    case 'UPDATE_WATCHED_VIDEOS': {
+      return {
+        ...state,
+        ...action.watchedVideos,
+      }
+    }
+    case 'UPDATE_FOLLOWED_CHANNELS': {
+      return {
+        ...state,
+        ...action.followedChannels,
+      }
+    }
+    default: {
+      throw new Error(`Unhandled action type: ${(action as any).type}`)
+    }
+  }
+}
+type ContextValue = {
+  state: PersonalData
+  dispatch: Dispatch<Action>
+}
+const PersonalDataContext = React.createContext<ContextValue | undefined>(undefined)
```

Comment:
Maybe it would be safer to initialize this with a default value? So we don't have to worry about `undefined` when accessing the `state`. That can be
```ts
const PersonalDataContext = React.createContext<ContextValue>({
  state: {
    followedChannels: [],
    watchedVideos: [],
  },
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  dispatch: (_) => {},
})
```